lms/djangoapps/course_api/api.py

```diff
@@ -19,6 +19,8 @@ def get_effective_user(requesting_user, target_username):
     """
     if target_username == requesting_user.username:
         return requesting_user
+    elif target_username == '' and requesting_user.username:
```

Comment:
The rationale was to specifically handle the case in which a logged-in user does not give a username, but I suppose this line would be fine if we have an anonymous user making the request as well. The effective user returned will be identical to the original anonymous user anyway.